bin/sdrfToNfConf.R

```diff
@@ -904,19 +905,22 @@ configs <- lapply(species_list, function(species){
     }
     
     # For droplet techs, add colums with strighforward statements of the URIs
-    # that contain barcodes and cDNAs, and record which columns to use
+    # that contain barcodes and cDNAs, and record which columns to use (unless
+    # HCA in which case the UUID added above will work
         
     if ( is.droplet.protocol(protocol)){
-     
+ 
       cdna_field <- getActualColnames('cdna read', sdrf)
       umi_field <- getActualColnames('umi barcode read', sdrf)
       cb_field <- getActualColnames('cell barcode read', sdrf)
-      uri_cols <- which(colnames(sdrf) == fastq.col)
-
-      if (length(uri_cols) < 2 ){
-        perror('Less than 3 FASTQ URI fields supplied for droplet experiment- expect one for each of cDNA, cell barcode and UMI')
-        q(status=1)
-      }    
+      
+      if (! is.hca){
+        uri_cols <- which(colnames(sdrf) == fastq.col)
+        if (length(uri_cols) < 2 ){
```

Comment:
Why do you check for less than 2 (so two would fail) but the message says less than 3... I guess the message is wrong?

Reply:
Thanks Pablo- error message fixed.